### DISCUSSION TITLE
Class 'Barryvdh\DomPDF\ServiceProvider' not found

### DISCUSSION THREAD

**Discussion by pherzing** [snipe/snipe-it#11090] [General]
Trying to upgrade a working 5.4.4 install to 6.0. I started with an upgrade of php from 7.2 to 7.4, which appears to be fine. I've verified that 7.4 being used by default server-wide, and by apache.

the usual php upgrade.php command includes a number of failures like this:

In ProviderRepository.php line 208:
Class 'Barryvdh\DomPDF\ServiceProvider' not found  

The upgrade appears to complete, but the web UI is a blank screen, nothing in the console. In apache logs I see things like:

`Uncaught ReflectionException: Class view does not exist`

I'm reverting to 5.4.4 for now, but if anyone has a clue as to where I've gone wrong, thank you.

**Comment by King-GO** [comment]
https://github.com/snipe/snipe-it/issues/11099

**Comment by snipe** [comment]
Duplicate of #11099

**Comment by savornicesei** [comment]
Run 
```bash
cd /{path where snipe-it is located}
composer update
#then probably you need php7-sodium
sudo zypper in php7-sodium #on openSUSE
php upgrade.php
```
All errors should be gone.

**Reply by pherzing** [reply]
Thank you, this was helpful. In my case there were some extra steps that others might find useful:

1. I'm running on a VM with only 1GB RAM, and I had to add swap (3GB, specifically) before composer update would run.
2. I also had to run composer self-update, to update composer from v1 to v2

My first php upgrade.php after this still resulted in the Barryvdh\DomPDF\ServiceProvider errors, but otherwise succeeded and snipeit ran. I am still investigating whether those errors are now addressed.
